### PR TITLE
Validate genesis block - Closes #2884

### LINKED ISSUE #2884: Create controller/schemas/genesisBlock.js to validate genesis block
### Expected behavior
The genesis block structure passed to the application must be validated for a valid structure. 

### Actual behavior
Currently we are not validating the genesis block 

### Which version(s) does this affect? (Environment, OS, etc...)
1.6.0

### PR DESCRIPTION
**Pull request by michielmulders**
### What was the problem?
No validation present for Genesis Block configuration when initializing the application.

### How did I fix it?
JSON Schema validation added.

### How to test it?
Start application or use an online schema validator.
Add the schema in it and test against all different genesis_block.json definitions.

### Review checklist

* The PR resolves #2884 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: framework/src/controller/schema/application.js [hunk 1]
```diff
@@ -6,8 +6,144 @@ module.exports = {
...
+
+	transactions: {
+		id: 'transactions',
+		type: 'object',
+		required: ['type', 'timestamp', 'senderPublicKey', 'signature'],
+		properties: {
+			type: {
+				type: 'integer',
+				minimum: 0,
+				maximum: 7,
```
Comment (nazarhussain) [inline framework/src/controller/schema/application.js]: In genesis block we can't have all transactions.

Comment (michielmulders) [reply framework/src/controller/schema/application.js]: Pushed new commit: enum that allows only type `0`, `2`, and `3`. (The types that occur in the `genesis_block.json`.

**Comment by nazarhussain** [comment]
@MaciejBaj We need to keep this in mind https://github.com/LiskHQ/lisk/pull/2888#discussion_r259390801 when we allow generation of genesis block for custom applications.

File: framework/src/modules/chain/helpers/z_schema.js [hunk 2]
```diff
@@ -393,6 +392,23 @@ const liskFormats = {
 			);
 		},
 	},
+	/**
+	 * Returns true if value is lower than or equal to `TOTAL_AMOUNT`.
+	 *
+	 * @param {Object} value
+	 * @returns {boolean}
+	 */
+	maxAmount: {
```
Comment (MaciejBaj) [inline framework/src/modules/chain/helpers/z_schema.js]: `maxAmount` would be `TOTAL_AMOUNT`. The validation is checking if it's a correct amount - positive and lower than `TOTAL_AMOUNT`. 

I suggest to rename it to `amount`.

Comment (michielmulders) [reply framework/src/modules/chain/helpers/z_schema.js]: Not possible to define both `string` and `number` for a custom schema.
I changed it to `numAmount` therefore as `amount` is already been taken by the amount validator for strings.

**Comment by MaciejBaj** [comment]
Also, we should address the issue of a different type for storing the amounts in genesis block with the goal to remove `numAmount` type from z-schema and change all of the amount occurrences across the application to strings. Can you open an issue for it @michielmulders?

**Pull request merged**